release: 0.9.0 (#383)
reanahub/reana-commons: CHANGES.rst

Changes
=======

> Version 0.9.0 (2022-12-13)

Reviewer: We should also add an entry for the workspace utils ([workspace: move from reana-workflow-controller](https://github.com/reanahub/reana-commons/commit/9cec0269e4cdfb1657019734edc42ba123770734) and following commits) and one entry to summarize all the changes we have made to r-server's OpenAPI (e.g. improved specification of the responses)

--------------------------

- Adds support for Python 3.11.
- Adds support for Rucio.
- Adds REANA specification validation and loading logic from ``reana-client``.

> - Adds support for networking/v1 API to Kubernetes Python client.

Reviewer: ```suggestion
- Changes the Kubernetes Python client to use  the ``networking/v1`` API.
```
WDYT?

- Adds OpenAPI specification support for ``launch`` endpoint that allows running workflows from remote sources.

> - Adds OpenAPI specification support for ``get_workflow_retention_rules`` endpoint.

Reviewer: ```suggestion
- Adds OpenAPI specification support for ``get_workflow_retention_rules`` endpoint that allows to retrieve the workspace file retention rules of a workflow.
```

- Adds support for specifying ``slurm_partition`` and ``slurm_time`` for Slurm compute backend jobs.
- Adds generation of Kerberos init and renew container's configuration.
- Adds specification of ``retention_days`` in ``reana.yaml`` JSON schema.
- Adds support for Unicode characters inside email body.
- Changes REANA specification loading functionality to allow specifying different working directories.

> - Changes REANA specification to allow enabling Kerberos for the whole workflow.

Reviewer: I think we should keep this consistent with  _Adds specification of ``retention_days`` in ``reana.yaml`` JSON schema._, as they make the same kind of changes to `reana.yaml` schema.

Maybe we can write something similar to _Adds support for specifying ``slurm_partition`` and ``slurm_time`` for Slurm compute backend jobs._ for both of them?

- Changes the loading of Snakemake specifications to preserve the current working directory.

> - Fixes job controller submission API to strip potential leading and trailing whitespaces in image names.

Reviewer: ```suggestion
- Fixes the submission of jobs by stripping potential leading and trailing whitespaces in Docker image names.
```
